### PR TITLE
fix(button-simple): added tabindex for accessibility

### PR DESCRIPTION
**Pull request by deepak-rohilla**
# Description

PR to add tabIndex prop to ButtonSimple for accessibility as the react-aria useButton doesn't forwards the received tabIndex prop and always returns undefined for it.

# Links

[*useButton tabIndex issue.*](https://github.com/adobe/react-spectrum/issues/2799)

### CODE REVIEW

**Review by cscheffauer — APPROVED** [review]

**Comment by torgeadelin** [comment]
There is a test failing, otherwise looks ok.

File: src/components/SearchInput/SearchInput.tsx [hunk 1]
```diff
@@ -77,6 +77,7 @@ const SearchInput = (props: Props, ref: RefObject<HTMLInputElement>): ReactEleme
           className={STYLE.clear}
           {...clearButtonProps}
           aria-label={clearButtonAriaLabel}
+          tabIndex={-1}
```
Comment (cscheffauer) [inline src/components/SearchInput/SearchInput.tsx:80]: this button inside of the searchInput should be focusable, so setting the tabIndex to -1 here i think isn't correct and has an effect on Cantina. 
@torgeadelin can u pls confirm the accessibility behaviour we are expecting from that clear button in searchInputs?

**Review by cscheffauer — CHANGES_REQUESTED** [review]

File: src/components/SearchInput/SearchInput.tsx (hunk 1 above)
Comment (deepak-rohilla) [reply src/components/SearchInput/SearchInput.tsx]: Have updated the tabIndex handling of ButtonSimple and removed the modifications from other files, thanks for the help @cscheffauer.

**Review by torgeadelin — APPROVED** [review]
lgtm

**Review by cscheffauer — APPROVED** [review]

**Pull request merged**